rdf4j/src/main/java/org/apache/commons/rdf/rdf4j/RDF4JGraphLike.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.rdf.rdf4j;
+
+import java.util.Optional;
+
+import org.apache.commons.rdf.api.BlankNodeOrIRI;
+import org.apache.commons.rdf.api.GraphLike;
+import org.apache.commons.rdf.api.IRI;
+import org.apache.commons.rdf.api.RDFTerm;
+import org.apache.commons.rdf.api.TripleLike;
+import org.eclipse.rdf4j.model.Model;
+import org.eclipse.rdf4j.repository.Repository;
+
+/**
+ * Marker interface for RDF4J implementations of GraphLike.
+ * 
+ * @see RDF4JGraph
+ * 
+ */
+public interface RDF4JGraphLike<T extends TripleLike<BlankNodeOrIRI, IRI, RDFTerm>>
```

Comment:
Yeah, it's tricky because Java does not have the equivalent of `owl:equivalentClass` and we wanted `TripleLike` to also be useful for generalized RDF (as in Jena - [createGeneralizedTriple](http://stain.github.io/incubator-commonsrdf/COMMONSRDF-35/org/apache/commons/rdf/jena/JenaRDFTermFactory.html#createGeneralizedTriple-org.apache.commons.rdf.api.RDFTerm-org.apache.commons.rdf.api.RDFTerm-org.apache.commons.rdf.api.RDFTerm-)).

so if we make:

```
public interface RDFTripleLike extends TripleLike<BlankNodeOrIRI, IRI, RDFTerm> {}
```

then that would simplify uses like `RDF4JGraphLike` (both Graph and Dataset requires the those RDF types for s/p/o), but it would mean you could not do "upcasting" from `TripleLike`:

```
TripleLike<BlankNodeOrIRI, IRI, RDFTerm> t; // get from outside
rdfThingie.write((RDFTripleLike)t);
```

So `TripleLike` would then be a bit more pointless than it is now for most users :)